sound/soc/sof/ipc4-pcm.c

```diff
@@ -901,7 +901,7 @@ static snd_pcm_sframes_t sof_ipc4_pcm_delay(struct snd_soc_component *component,
 	}
 
 	if (head_ptr < tail_ptr)
-		return substream->runtime->boundary - tail_ptr + head_ptr;
+		return substream->runtime->buffer_size - tail_ptr + head_ptr;
```

Comment:
This begs the question "how is the boundary taken into account"?

This looks like a controversial change without much justification. That is unlikely to be accepted upstream....

Reply:
boundary is used for app_ptr and hw_ptr wrapping. Now we use snd_sof_pcm_platform_pointer which is wrapped with runtime->buffer_size, so we need to wrap be dai link position with runtime->buffer_size so that we can compare the snd_sof_pcm_platform_pointer result with be dai link position